frontend-react/src/layouts/Markdown/LayoutComponents.tsx

```diff
@@ -0,0 +1,25 @@
+import { useContext, useEffect } from "react";
+
+import MarkdownLayoutContext from "./Context";
+
+export const LayoutSidenav = ({ children }: { children: React.ReactNode }) => {
+    const { setSidenavContent } = useContext(MarkdownLayoutContext);
+
+    useEffect(() => {
+        setSidenavContent(children);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return null;
+};
+
+export const LayoutMain = ({ children }: { children: React.ReactNode }) => {
+    const { setMainContent } = useContext(MarkdownLayoutContext);
+
+    useEffect(() => {
+        setMainContent(children);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we must be able to disable this on an app level...it's fully encouraged and supported by React.

Reply:
Like any eslint rule it can be changed, but didn't want to assume it should be changed in this PR.